packages/block-editor/src/components/list-view/use-list-view-images.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+// Maximum number of images to display in a list view row.
+const MAX_IMAGES = 5;
+
+function getImageUrl( block ) {
+	if ( block.name !== 'core/image' ) {
+		return;
+	}
+
+	if ( block.attributes?.url ) {
+		return { url: block.attributes.url, alt: block.attributes.alt };
+	}
+}
+
+function getImagesFromGallery( block ) {
+	if ( block.name !== 'core/gallery' || ! block.innerBlocks ) {
```

Comment:
Could this check be `if ( block.name !== 'core/gallery' && ! block.innerBlocks ) {` if it's just for galleries with innerBlocks?

Reply:
This is slightly subtle, but if we switched to `if ( block.name !== 'core/gallery' && ! block.innerBlocks ) {` in this conditional, then Group blocks would proceed to be processed.

Or do you mean refactor the function so that the innards of this function are placed within the `if` block?

Reply:
Never mind me. Logic malfunction.